src/pkg/project/manager.go

```diff
@@ -43,14 +45,22 @@ type Manager interface {
 	Get(ctx context.Context, idOrName interface{}) (*models.Project, error)
 
 	// List projects according to the query
-	List(ctx context.Context, query ...*models.ProjectQueryParam) ([]*models.Project, error)
+	List(ctx context.Context, query *q.Query) ([]*models.Project, error)
 }
 
 // New returns a default implementation of Manager
 func New() Manager {
 	return &manager{dao: dao.New()}
 }
 
+const projectNameMaxLen int = 255
+const projectNameMinLen int = 1
+const restrictedNameChars = `[a-z0-9]+(?:[._-][a-z0-9]+)*`
```

Comment:
is it possible to use the OCI defined reg string?

Reply:
In my impression, the `NameRegexp` of OCI includes `/` which is not valid for the name of the project.